File: app/src/main/java/org/deiverbum/app/data/entity/relation/MassReadingWithBook.java
package org.deiverbum.app.data.entity.relation;

import androidx.room.Embedded;
import androidx.room.Relation;

import org.deiverbum.app.data.entity.BibleReadingEntity;
import org.deiverbum.app.data.entity.BiblieBookEntity;
import org.deiverbum.app.model.MassReading;

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2023.1
 */
public class MassReadingWithBook {
    @Embedded
    public BibleReadingEntity lectura;

    @Relation(
            parentColumn = "bookFK",
            entityColumn = "bookID",
            entity = BiblieBookEntity.class
    )
    public BiblieBookEntity libro;

    public MassReading getDomainModel() {
        MassReading theModel = new MassReading();
        theModel.setLibro(libro.getDomainModel());
        theModel.setCapitulo(String.valueOf(lectura.getCapitulo()));
        theModel.setVersoInicial(String.valueOf(lectura.getDesde()));
        theModel.setVersoFinal(String.valueOf(lectura.getHasta()));
        theModel.setCita(lectura.getCita());
        theModel.setTexto(lectura.getTexto());
        return theModel;
    }

    public MassReading getDomainModelMisa() {
        MassReading theModel = new MassReading();
        theModel.setLibro(libro.getDomainModel());
        theModel.setCapitulo(String.valueOf(lectura.getCapitulo()));
        theModel.setVersoInicial(String.valueOf(lectura.getDesde()));
        theModel.setVersoFinal(String.valueOf(lectura.getHasta()));
        theModel.setCita(lectura.getCita());
        theModel.setTexto(lectura.getTexto());
        return theModel;
    }
}

File: app/src/main/java/org/deiverbum/app/data/entity/relation/MassReadingWithBook.kt
package org.deiverbum.app.data.entity.relation

import androidx.room.Embedded
import androidx.room.Relation
import org.deiverbum.app.data.entity.BibleReadingEntity
import org.deiverbum.app.data.entity.BibleBookEntity
import org.deiverbum.app.model.MassReading

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2023.1
 */
class MassReadingWithBook {
    @JvmField
    @Embedded
    var lectura: BibleReadingEntity? = null

    @JvmField
    @Relation(parentColumn = "bookFK", entityColumn = "bookID", entity = BibleBookEntity::class)
    var libro: BibleBookEntity? = null
    val domainModel: MassReading
        get() {
            val theModel = MassReading()
            theModel.book = libro!!.domainModel
            theModel.verseChapter = lectura!!.capitulo.toString()
            theModel.verseFrom = lectura!!.desde.toString()
            theModel.verseTo = lectura!!.hasta.toString()
            theModel.setCita(lectura!!.cita)
            theModel.text = lectura!!.texto
            return theModel
        }
    val domainModelMisa: MassReading
        get() {
            val theModel = MassReading()
            theModel.book = libro!!.domainModel
            theModel.verseChapter = lectura!!.capitulo.toString()
            theModel.verseFrom = lectura!!.desde.toString()
            theModel.verseTo = lectura!!.hasta.toString()
            theModel.setCita(lectura!!.cita)
            theModel.text = lectura!!.texto
            return theModel
        }
}